src/kudu/consensus/raft_consensus.cc

```diff
@@ -3688,9 +3688,12 @@ RaftConsensus::ChangeVoterDistribution(const TopologyConfigPB &topology_config)
                peer_uuid(), "tablet", options_.tablet_id);
   ThreadRestrictions::AssertWaitAllowed();
   LockGuard l(lock_);
-  // Do not allow any voter distribution changes on a unsquelched ring.
-  Status s = CheckNoConfigChangePendingUnlocked();
-  RETURN_NOT_OK(s);
+
+  if (adjust_voter_distribution_) {
```

Comment:
we should leave a comment here something like this for future reference.

"There are 2 use cases in which we ChangeVoterDistribution. One is the common
case when new regions are added or when the number of voters in a region is increasing/shrinking.

The other case is during Unsafe Config Change where we explicitly have to reduce
voter distribution to restore availability"